packages/core/src/ui/components/WalletSetupRevamp/WalletSetupReuseMnemonicStep.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { WalletSetupStepLayoutRevamp } from './WalletSetupStepLayoutRevamp';
+import { WalletTimelineSteps } from '@ui/components/WalletSetup';
+import { Box, Flex, Select, Text } from '@input-output-hk/lace-ui-toolkit';
+import { AnyWallet } from '@cardano-sdk/web-extension';
+import { Wallet } from '@lace/cardano';
+import { useTranslation } from 'react-i18next';
+
+type WalletSetupReuseMnemonicStepProps = {
+  wallets: AnyWallet<Wallet.WalletMetadata, Wallet.AccountMetadata>[];
+  onSelectedWalletChange?: (wallet: AnyWallet<Wallet.WalletMetadata, Wallet.AccountMetadata> | undefined) => void;
+  onBack: () => void;
+  onNext: () => void;
+};
+
+export const WalletSetupReuseMnemonicStep = ({
+  wallets = [],
+  onSelectedWalletChange,
+  onNext,
+  onBack
+}: WalletSetupReuseMnemonicStepProps): ReactElement => {
+  const [selectedWallet, setSelectedWallet] = useState<string | undefined>(wallets[0]?.walletId);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    setSelectedWallet(wallets[0]?.walletId);
+    onSelectedWalletChange?.(wallets[0]);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [wallets]);
```

Comment:
```suggestion
  }, [wallets, onSelectedWalletChange]);
```

and just pass `setWalletToReuse` directly into the `WalletSetupReuseMnemonicStep` in `ReuseRecoveryPhrase`